server/server.go

```diff
@@ -1002,42 +1002,73 @@ func (s *Server) getClientConnectURLs() []string {
 	sPort := strconv.Itoa(opts.Port)
 	urls := make([]string, 0, 1)
 
-	ipAddr, err := net.ResolveIPAddr("ip", opts.Host)
-	// If the host is "any" (0.0.0.0 or ::), get specific IPs from available
-	// interfaces.
-	if err == nil && ipAddr.IP.IsUnspecified() {
-		var ip net.IP
-		ifaces, _ := net.Interfaces()
-		for _, i := range ifaces {
-			addrs, _ := i.Addrs()
-			for _, addr := range addrs {
-				switch v := addr.(type) {
-				case *net.IPNet:
-					ip = v.IP
-				case *net.IPAddr:
-					ip = v.IP
-				}
-				// Skip non global unicast addresses
-				if !ip.IsGlobalUnicast() || ip.IsUnspecified() {
-					ip = nil
-					continue
+	// short circuit if cluster-advertise is set
+	if opts.Cluster.ClientAdvertiseStr != "" {
+		hosts := strings.Split(opts.Cluster.ClientAdvertiseStr, ",")
+
+		for n, i := range hosts {
```

Comment:
Bad choice for variables. `i` is often used in for loops as the index in the array, not the element. I would replace with `a` for address or `addr` and could get rid of `n` (see below).